atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -959,6 +961,17 @@ private int estimateSize(List<RowResult<Value>> rangeRows) {
             }
         }
 
+        if (AtlasDbConstants.hiddenTables.contains(tableRef)) {
```

Comment:
So all hidden tables are actually used non-transactionally? Make sense.

So, if someone adds a new hidden table that is transactional, we will mistakenly exclude it from postfiltering. On the other hand, if we make a separate list of non-transactional tables, then someone will probably forget his new table to the list. Either way, this seems quite fragile.

A more robust solution would be to treat the timestamp 0 as always committed and visible by any transaction, but that is probably more tricky to implement correctly. Your solution makes sense as a quick fix to the problem. 👍